corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -329,7 +329,8 @@ def as_string(self):
             blob.close()
 
     def get_http_response(self):
-        return stream_response(self.blobdb.get(self.get_filename()))
+        headers = {'Content-Length': self.blobdb.size(self.get_filename())}
```

Comment:
It is unfortunate that we need to make two requests to Riak CS, one for content and one for size. Presumably, there is a content-length header returned from the original request for content, which could be reused.

Relatedly, it would be even better to use a proxy header like [`XSendfile`](https://www.nginx.com/resources/wiki/start/topics/examples/xsendfile/) to serve the blob directly from Riak CS (s3db) or file system (fsdb) rather than relaying through Django like this.